### DISCUSSION TITLE
Flashing issue with different cfg file

### DISCUSSION THREAD

**Discussion by laurenbramblett** [bitcraze/discussions#674] [Q&A]
Hello,

I am trying to flash the gap8 bootloader with a JTAG 20-10 connector and an ST-Link-v2. I know that the ST-Link-v2 is JTAG enabled according to its documentation and supported by openOCD, but when I try the following command:
docker run --rm -it -v $PWD:/module/ --device /dev/ttyUSB0 --privileged -P bitcraze/aideck /bin/bash -c 'export GAPY_OPENOCD_CABLE=interface/stlink.cfg; source /gap_sdk/configs/ai_deck.sh; cd /module/;  make all image flash'

I get the following errors:
gapy --target=ai_deck --platform=board --work-dir=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --image --binary=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/bootloader 
gapy --target=ai_deck --platform=board --work-dir=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --flash --binary=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/bootloader 
make: Nothing to be done for 'image'.
gapy --target=ai_deck --platform=board --work-dir=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --flash --force --binary=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/bootloader 
Flashing image with command:
gap8-openocd -d0 -c "gdb_port disabled; telnet_port disabled; tcl_port disabled" -c "script interface/ftdi/stlink.cfg; script target/gap8revb.tcl; script tcl/flash_image.tcl; script tcl/jtag_boot.tcl; gap_flash_raw_hyper /module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/target.board.devices.flash.img 75808 /gap_sdk/utils/openocd_tools; exit;"
Open On-Chip Debugger 0.10.0+dev-gb84d97ec4 (2022-04-20-14:26)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
debug_level: 0

embedded:startup.tcl:26: Error: Can't find interface/ftdi/stlink.cfg
in procedure 'script' 
at file "embedded:startup.tcl", line 26
Traceback (most recent call last):
  File "/gap_sdk/utils/gapy/gapy", line 162, in <module>
    main()
  File "/gap_sdk/utils/gapy/gapy", line 152, in main
    operationFunc(args, config, system)
  File "/gap_sdk/utils/gapy/run.py", line 184, in operationFunc
    raise RuntimeError('Runner has failed with value: %d' % status)
RuntimeError: Runner has failed with value: -1
make: *** [/gap_sdk/rtos/freeRTOS/vendors/gwt/rules/freeRTOS_rules.mk:404: flash] Error 1

What am I doing wrong? 
Thank you for your help!

Lauren

**Comment by ataffanel** [comment]
Hi,

The important part of the error is `embedded:startup.tcl:26: Error: Can't find interface/ftdi/stlink.cfg`, the file `interface/ftdi/stlink.cfg` does not exist, it is not the right path. It should instead be `interface/stlink.cfg`.

**Comment by laurenbramblett** [comment]
I apologize, I copied the wrong error code. That was just one of my iterations. Here it is with "interface/stlink.cfg":
`docker run --rm -it -v $PWD:/module/ --device /dev/ttyUSB0 --privileged -P bitcraze/aideck /bin/bash -c 'export GAPY_OPENOCD_CABLE=interface/stlink.cfg; source /gap_sdk/configs/ai_deck.sh; cd /module/;  make all image flash'`

Gives the following error:
```
gapy --target=ai_deck --platform=board --work-dir=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --image --binary=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/bootloader 
gapy --target=ai_deck --platform=board --work-dir=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --flash --binary=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/bootloader 
make: Nothing to be done for 'image'.
gapy --target=ai_deck --platform=board --work-dir=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --flash --force --binary=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/bootloader 
Flashing image with command:
gap8-openocd -d0 -c "gdb_port disabled; telnet_port disabled; tcl_port disabled" -c "script interface/stlink.cfg; script target/gap8revb.tcl; script tcl/flash_image.tcl; script tcl/jtag_boot.tcl; gap_flash_raw_hyper /module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/target.board.devices.flash.img 75808 /gap_sdk/utils/openocd_tools; exit;"
Open On-Chip Debugger 0.10.0+dev-gb84d97ec4 (2022-04-20-14:26)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
debug_level: 0

DEPRECATED! use 'adapter speed' not 'adapter_khz'
DEPRECATED! use 'adapter driver' not 'interface'
Error: Debug adapter doesn't support 'jtag' transport
/gap_sdk/install/workstation/gap8-openocd/share/openocd/scripts/target/gap8revb.tcl:11: Error: 
in procedure 'script' 
at file "embedded:startup.tcl", line 26
at file "/gap_sdk/install/workstation/gap8-openocd/share/openocd/scripts/target/gap8revb.tcl", line 11
Traceback (most recent call last):
  File "/gap_sdk/utils/gapy/gapy", line 162, in <module>
    main()
  File "/gap_sdk/utils/gapy/gapy", line 152, in main
    operationFunc(args, config, system)
  File "/gap_sdk/utils/gapy/run.py", line 184, in operationFunc
    raise RuntimeError('Runner has failed with value: %d' % status)
RuntimeError: Runner has failed with value: -1
make: *** [/gap_sdk/rtos/freeRTOS/vendors/gwt/rules/freeRTOS_rules.mk:404: flash] Error 1
```

**Reply by ataffanel** [reply]
I am not sure why you are getting this error, the STLink is supposed to support `jtag`.

In out tests, we never managed to flash the GAP-8 with an STLink, we only managed with an Olimex programming probe or a J-Link (see https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#gap8-bootloader-for-the-aideck-1-0-or-aideck-1-1-rev-a-c-). So it is likely you will need to find another programming probe in order to flash the gap8.
